File: 26-RemoveDuplicatesFromSortedArray/26-RemoveDuplicatesFromSortedArray.java
// Last updated: 16/10/2025, 04:31:09
class Solution {
    public int removeDuplicates(int[] nums) {
        int n = nums.length;
        int s = 0;
        int e = 0;

        while(s < n){
            if(nums[s] == nums[e]){
                s++;
            }
            else{
                nums[++e] = nums[s++];
            }
        }
        return e + 1;
    }
}

File: 26-RemoveDuplicatesFromSortedArray/26-RemoveDuplicatesFromSortedArray.cpp
// Last updated: 16/10/2025, 04:31:16
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int n = nums.size();
        int j = 1;
        int i=0;
        while(i < n-1){
            if(nums[i] != nums[i+1]){
                nums[j++] = nums[i+1];
            }
            i++;
        }
        return j;
    }
};

 